### PR TITLE
Better support for sql outside a database directory

### PR DESCRIPTION
**Pull request by zachmu**
This change provides a fallback user name and email to use for creating databases in SQL.

Also fixed bugs in config management and added tests.

### CODE REVIEW

File: go/cmd/dolt/commands/config.go [hunk 1]
```diff
@@ -142,32 +142,35 @@ func getOperation(dEnv *env.DoltEnv, setCfgTypes *set.StrSet, args []string, pri
...
+	case 1:
+		configElement := newCfgElement(setCfgTypes.AsSlice()[0])
+		var ok bool
+		cfg, ok = dEnv.Config.GetConfig(configElement)
+		if !ok {
+			cli.Println(color.RedString("No config found for %s", configElement.String()))
 			return 1
 		}
+	default:
+		// should be impossible due to earlier checks
```
Comment (max-hoffman) [inline go/cmd/dolt/commands/config.go:158]: is there any point to `StrSet` if we're not allowed to pass multiple? if `[]{'local', 'global'}` if being pre handled earlier just pass the string right?

File: integration-tests/bats/bats.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+select "                    _..-'(                       )`-.._\n                   ./'. '||\\\\.       (\\_/)       .//||` .`\\.\n                ./'.|'.'||||\\\\|..    )O O(    ..|//||||`.`|.`\\.\n             ./'..|'.|| |||||\\`````` '`\"'` ''''''/||||| ||.`|..`\\.\n           ./'.||'.|||| ||||||||||||.     .|||||||||||| |||||.`||.`\\.\n          /'|||'.|||||| ||||||||||||{     }|||||||||||| ||||||.`|||`\\\n         '.|||'.||||||| ||||||||||||{     }|||||||||||| |||||||.`|||.`\n        '.||| ||||||||| |/'   ``\\||``     ''||/''   `\\| ||||||||| |||.`\n        |/' \\./'     `\\./         \\!|\\   /|!/         \\./'     `\\./ `\\|\n        V    V         V          }' `\\ /' `{          V         V    V\n        `    `         `               V               '         '    '\n" as "BATS!";
```
Comment (max-hoffman) [inline integration-tests/bats/bats.sql:1]: hmm

File: integration-tests/bats/config.bats [hunk 3]
```diff
@@ -179,15 +213,49 @@ teardown() {
...
+    [ "$status" -eq 0 ]
+
+    run dolt log -n 1
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "bats tester" ]] || false
+    [[ "$output" =~ "doltuser@dolthub.com" ]] || false
+    [[ "$output" =~ "created table test2" ]] || false
 }
 
+
```
Comment (max-hoffman) [inline integration-tests/bats/config.bats:258]: nit delete extra line

File: integration-tests/bats/sql.bats [hunk 4]
```diff
@@ -646,8 +646,29 @@ SQL
...
+    skip "This results in a panic right now"
+    
+    run dolt sql -q "show databases" -r csv
+    [ "$status" -eq 0 ]
+    [ "${#lines[@]}" -eq 2 ]
+    [[ "$output" =~ "information_schema" ]] || false
+    [[ ! "$output" =~ "decoy" ]] || false
 }
 
+
```
Comment (max-hoffman) [inline integration-tests/bats/sql.bats:671]: nit extra line

**Review by max-hoffman — APPROVED** [review]
LGTM couple nits. i expected the PR to make DoltCliConfig implement `WriteableConfig` and `ReadWriteConfig` but can't think of any issues with this

File: go/cmd/dolt/commands/config.go (hunk 1 above)
Comment (zachmu) [reply go/cmd/dolt/commands/config.go:158]: No real point, but it's there for symmetry with the other methods (needed by list). Didn't want to mess with it any more

File: go/libraries/doltcore/env/config.go [hunk 5]
```diff
@@ -230,25 +229,56 @@ type writeableLocalDoltCliConfig struct {
...
 	return writeableLocalDoltCliConfig{dcc}
 }
 
+// SetFailsafes sets the config values given as failsafes, i.e. values that will be returned as a last resort if they
+// are not found elsewhere in the config hierarchy. The "failsafe" config can be written to in order to conform to the
+// interface of ConfigHierarchy, but values will not persist beyond this session.
+// Calling SetFailsafes more than once will overwrite any previous values.
+// Should only be called after primary configuration of the config hierarchy has been completed.
+func (dcc DoltCliConfig) SetFailsafes(cfg map[string]string) {
+	existing, ok := dcc.ch.GetConfig("failsafe")
```
[the remark this answers is not in the document]
Comment (zachmu) [reply go/libraries/doltcore/env/config.go:243]: Nah, I'm not super sure about this interface

File: integration-tests/bats/bats.sql (hunk 2 above)
Comment (zachmu) [reply integration-tests/bats/bats.sql:1]: Assume you ran it

Comment (max-hoffman) [reply integration-tests/bats/bats.sql:1]: :) couldn't resist

**Pull request merged**